### PR TITLE
Add a subset of LFW dataset to `skimage.data`

### PR DESCRIPTION
**Pull request by glemaitre**
## Description
[Tell us about your new feature, improvement, or fix! If relevant, please supplement the PR with images.]

## Checklist
[It's fine to submit PRs which are a work in progress! But before they are merged, all PRs should provide:]
- [x] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- [x] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [x] Gallery example in `./doc/examples` (new features only)
- [x] Unit tests

[For detailed information on these and other aspects see [scikit-image contribution guidelines](http://scikit-image.org/docs/dev/contribute.html)]

## References
[If this is a bug-fix or enhancement, it closes issue # ]
[If this is a new feature, it implements the following paper: ]

## For reviewers

(Don't remove the checklist below.)

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

File: skimage/data/tests/test_data.py [hunk 1]
```diff
@@ -99,3 +99,8 @@ def test_binary_blobs():
 def test_cbcl_face_database():
     """ Test that "cbcl_face_database" can be loaded."""
     data.cbcl_face_database()
+
+
+def test_lfw_subset():
+    """ Test that "lfw_subset" can be loaded."""
+    data.lfw_subset
```
Comment (massich) [inline skimage/data/tests/test_data.py]: nice !!!

**Comment by stefanv** [comment]
Conflicts, otherwise looks good.

**Comment by glemaitre** [comment]
The failure does not seem related

**Comment by glemaitre** [comment]
@stefanv I noticed that this is MRG + 2, is it ready to be merged. Once done, I could finalize the Haar example.

**Comment by glemaitre** [comment]
ping @stefanv @soupault @warmspringwinds Could we go ahead with the PR such that I can update the example.

**Pull request merged**